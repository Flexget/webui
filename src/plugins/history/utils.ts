import { groupBy } from 'utils/array';
import { HistoryItem, GroupByFields } from './types';

export const getGroupedHistory = (items: HistoryItem[], grouping: GroupByFields) =>
  groupBy(
    items,
    grouping,
    grouping === GroupByFields.Time ? (i: string) => i.split('T')[0] : undefined,
  );
