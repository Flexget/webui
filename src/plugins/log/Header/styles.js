import styled from '@emotion/styled';
import { css } from '@emotion/core';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import theme from 'theme';

import { Typography, TextField } from '@material-ui/core';

const grey = css`
  color: ${theme.palette.grey[600]};
`;

export const Wrapper = styled.div`
  display: block;
  ${theme.breakpoints.up('sm')} {
    display: flex;
    align-items: baseline;
  }
`;

export const GreyIcon = styled(FontAwesomeIcon)`
  ${grey};
`;

export const GreyClickableIcon = styled(GreyIcon)`
  cursor: pointer;
`;

export const GreyType = styled(Typography)`
  ${grey};
`;

export const Spacer = styled.div`
  flex: 1;
`;

export const TextFieldWrapper = styled.div`
  display: flex;
  align-items: baseline;
`;

export const FilterField = styled(TextField)`
  margin-left: 1rem;
  margin-right: 1rem;
`;

export const MenuIcon = styled(FontAwesomeIcon)`
  margin-right: 3rem;
`;
