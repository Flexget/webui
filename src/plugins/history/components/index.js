import { connect } from 'react-redux';
import { request } from 'utils/actions';
import { GET_HISTORY } from 'plugins/history/data/actions';
import History from './History';

function mapDispatchToProps(dispatch) {
  return {
    getHistory: data => dispatch(request(GET_HISTORY, data)),
  };
}

export default connect(null, mapDispatchToProps)(History);
export { History };
