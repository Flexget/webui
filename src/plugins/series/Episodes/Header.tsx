import React, { FC, ChangeEvent, useEffect } from 'react';
import { TablePagination, Theme } from '@material-ui/core';
import { css, ClassNames } from '@emotion/core';
import SelectField from 'common/inputs/formik/SelectField';
import { useContainer } from 'unstated-next';
import { useFormikContext } from 'formik';
import { useDebounce } from 'utils/hooks';
import { Direction } from 'utils/query';
import { EpisodeContainer } from '../hooks/episodes';
import { GetEpisodeOptions } from '../types';

interface Props {
  options: GetEpisodeOptions;
}

const wrapper = (theme: Theme) => css`
  display: flex;
  justify-content: space-between;
  padding-bottom: ${theme.typography.pxToRem(theme.spacing(2))};
  align-items: center;
`;

const container = css`
  display: flex;
  align-items: center;
  font-size: 14px;
`;

const item = (theme: Theme) => css`
  margin-left: ${theme.typography.pxToRem(theme.spacing(2))};
  margin-right: ${theme.typography.pxToRem(theme.spacing(2))};
`;

const input = css`
  font-size: inherit;
`;

const orderByOptions = [
  {
    value: Direction.Desc,
    label: 'Desc',
  },
  {
    value: Direction.Asc,
    label: 'Asc',
  },
];

const Header: FC<Props> = ({ options: { page, perPage } }) => {
  const [{ totalCount }] = useContainer(EpisodeContainer);

  const { values, submitForm, setFieldValue } = useFormikContext<GetEpisodeOptions>();

  const handleChangePerPage = (event: ChangeEvent<HTMLInputElement>) => {
    setFieldValue('perPage', parseInt(event.target.value, 10));
    setFieldValue('page', 0);
  };

  const handleChangePage = (_: unknown, p: number) => {
    setFieldValue('page', p);
  };

  const debouncedValues = useDebounce(values);

  useEffect(() => {
    submitForm();
  }, [...Object.values(debouncedValues), submitForm]); // eslint-disable-line react-hooks/exhaustive-deps

  return (
    <>
      <ClassNames>
        {({ css: cssString }) => (
          <div css={wrapper}>
            <div css={container} />
            <div css={container}>
              <SelectField
                css={item}
                label="Order"
                name="order"
                id="order"
                size="small"
                InputProps={{ className: cssString(input) }}
                options={orderByOptions}
              />
              <TablePagination
                rowsPerPageOptions={[30, 60, 90]}
                count={totalCount}
                rowsPerPage={perPage}
                page={page}
                onChangePage={handleChangePage}
                onChangeRowsPerPage={handleChangePerPage}
                component="div"
              />
            </div>
          </div>
        )}
      </ClassNames>
    </>
  );
};

export default Header;
