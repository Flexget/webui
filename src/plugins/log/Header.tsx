import React, {
  FC,
  useState,
  useCallback,
  MouseEvent,
  useMemo,
  ComponentType,
  useEffect,
} from 'react';
import { useFormikContext } from 'formik';
import {
  Theme,
  Typography,
  IconButton,
  Menu,
  MenuItem,
  ListItemIcon,
  CircularProgress,
} from '@material-ui/core';
import { css } from '@emotion/core';
import { Spacer } from 'common/styles';
import TextField from 'common/inputs/formik/TextField';
import { PlayArrow, Stop, FilterList, MoreVert, ClearAll } from '@material-ui/icons';
import { ReadyState } from 'core/api';
import { useDebounce } from 'utils/hooks';
import { Options } from './types';

export const wrapper = (theme: Theme) => css`
  display: block;
  ${theme.breakpoints.up('sm')} {
    display: flex;
    align-items: baseline;
  }
`;

export const filterWrapper = css`
  display: flex;
  align-items: center;
`;

export const filterField = css`
  margin-left: 1rem;
  margin-right: 1rem;
`;

interface Props {
  readyState: ReadyState;
  connect: () => void;
  disconnect: () => void;
  clear: () => void;
}

interface StateOptions {
  onClick?: () => void;
  Icon: ComponentType;
  label: string;
  heading: string;
  disabled?: boolean;
}

const Header: FC<Props> = ({ readyState, connect, disconnect, clear }) => {
  const [anchorEl, setAnchorEl] = useState<HTMLButtonElement>();
  const helperText = 'Supports operators and, or, (), and "str"';

  const stateMap: Record<ReadyState, StateOptions> = useMemo(
    () => ({
      [ReadyState.Open]: {
        onClick: disconnect,
        Icon: Stop,
        label: 'Stop',
        heading: 'Streaming',
      },
      [ReadyState.Connecting]: {
        label: 'Start',
        Icon: CircularProgress,
        heading: 'Connecting',
        disabled: true,
      },
      [ReadyState.Closed]: {
        onClick: connect,
        Icon: PlayArrow,
        label: 'Start',
        heading: 'Disconnected',
      },
    }),
    [connect, disconnect],
  );

  const { onClick, Icon, heading, label, disabled = false } = stateMap[readyState];

  const handleMenuClose = useCallback(() => setAnchorEl(undefined), []);
  const handleMenuOpen = useCallback(
    (event: MouseEvent<HTMLButtonElement>) => setAnchorEl(event.currentTarget),
    [],
  );

  const { values, submitForm } = useFormikContext<Options>();

  const debouncedValues = useDebounce(values);

  useEffect(() => {
    submitForm();
  }, [...Object.values(debouncedValues), submitForm]);

  return (
    <div css={wrapper}>
      <div>
        <Typography variant="h6">Server Log</Typography>
        <Typography variant="subtitle1" color="textSecondary">
          {heading}
        </Typography>
      </div>
      <Spacer />
      <div css={filterWrapper}>
        <FilterList />
        <TextField
          css={filterField}
          id="query"
          name="query"
          label="Filter"
          helperText={helperText}
        />
        <IconButton onClick={handleMenuOpen}>
          <MoreVert />
        </IconButton>
      </div>
      <Menu id="log-menu" anchorEl={anchorEl} open={!!anchorEl} onClose={handleMenuClose}>
        <MenuItem>
          <TextField id="lines" name="lines" label="Max Lines" type="number" />
        </MenuItem>
        <MenuItem onClick={clear}>
          <ListItemIcon>
            <ClearAll />
          </ListItemIcon>
          Clear
        </MenuItem>
        <MenuItem onClick={onClick} disabled={disabled}>
          <ListItemIcon>
            <Icon />
          </ListItemIcon>
          {label}
        </MenuItem>
      </Menu>
    </div>
  );
};

export default Header;
