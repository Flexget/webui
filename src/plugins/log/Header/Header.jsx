import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Menu, MenuItem, TextField, Typography, IconButton, ListItemIcon } from '@material-ui/core';
import { FilterList, MoreVert, Stop, PlayArrow, ClearAll } from '@material-ui/icons';
import { Wrapper, Spacer, TextFieldWrapper, FilterField } from './styles';

const ENTER_KEY = 13;

class Header extends Component {
  static propTypes = {
    start: PropTypes.func.isRequired,
    stop: PropTypes.func.isRequired,
    connected: PropTypes.bool.isRequired,
    clearLogs: PropTypes.func.isRequired,
  };

  state = {
    open: false,
    lines: '200',
    query: '',
  };

  componentDidMount() {
    this.reload();
  }

  componentWillUnmount() {
    const { stop } = this.props;
    stop();
  }

  reload = () => {
    const { connected, start, stop } = this.props;
    const { lines, query } = this.state;

    if (connected) {
      stop();
    }
    start({ lines, query });
  };

  clearLogs = () => this.props.clearLogs();

  handleLines = event => this.setState({ lines: event.target.value });

  handleQuery = event => this.setState({ query: event.target.value });

  handleKeyPress = event => event.which === ENTER_KEY && this.reload();

  handleRequestClose = () =>
    this.setState({
      open: false,
      anchorEl: undefined,
    });

  handleMenuClick = event =>
    this.setState({
      open: true,
      anchorEl: event.currentTarget,
    });

  render() {
    const { connected, stop } = this.props;
    const { anchorEl, open, query, lines } = this.state;
    const helperText = 'Supports operators and, or, (), and "str"';
    const { onClick, Icon, text } = connected
      ? {
          onClick: stop,
          Icon: Stop,
          text: 'Stop',
        }
      : {
          onClick: this.reload,
          Icon: PlayArrow,
          text: 'Start',
        };

    return (
      <Wrapper>
        <div>
          <Typography variant="h6">Server Log</Typography>
          <Typography variant="subtitle1" color="textSecondary">
            {connected ? 'Streaming' : 'Disconnected'}
          </Typography>
        </div>
        <Spacer />
        <TextFieldWrapper>
          <FilterList />
          <FilterField
            id="filter"
            label="Filter"
            value={query}
            onChange={this.handleQuery}
            inputProps={{
              onKeyPress: this.handleKeyPress,
            }}
            helperText={helperText}
          />
          <IconButton onClick={this.handleMenuClick}>
            <MoreVert />
          </IconButton>
        </TextFieldWrapper>
        <Menu id="log-menu" anchorEl={anchorEl} open={open} onClose={this.handleRequestClose}>
          <MenuItem>
            <TextField
              id="lines"
              label="Max Lines"
              value={lines}
              onChange={this.handleLines}
              type="number"
              inputProps={{
                onKeyPress: this.handleKeyPress,
              }}
            />
          </MenuItem>
          <MenuItem onClick={this.clearLogs}>
            <ListItemIcon>
              <ClearAll />
            </ListItemIcon>
            Clear
          </MenuItem>
          <MenuItem onClick={onClick}>
            <ListItemIcon>
              <Icon />
            </ListItemIcon>
            {text}
          </MenuItem>
        </Menu>
      </Wrapper>
    );
  }
}

export default Header;
