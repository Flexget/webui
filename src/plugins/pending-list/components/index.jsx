import React, { PureComponent } from 'react';
import { hot } from 'react-hot-loader';
import { NoPaddingWrapper } from 'components/styles';
import TabList from './TabList';

class PendingList extends PureComponent {
  state = {
    listId: false,
  }

  selectList = listId => this.setState({ listId })

  render() {
    const { listId } = this.state;

    return (
      <NoPaddingWrapper>
        <TabList
          listId={listId}
          selectList={this.selectList}
        />
      </NoPaddingWrapper>
    );
  }
}

export default hot(module)(PendingList);
