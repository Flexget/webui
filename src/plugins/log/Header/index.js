import { connect } from 'react-redux';
import actions from 'plugins/log/state/actions';
import Header from './Header';

export function mapStateToProps({ log }) {
  return {
    connected: log.connected,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    start: payload => dispatch(actions.connect.request(payload)),
    stop: () => dispatch(actions.disconnect()),
    clearLogs: () => dispatch(actions.clear()),
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Header);
export { Header };
