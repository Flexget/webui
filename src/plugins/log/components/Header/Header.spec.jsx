import React from 'react';
import renderer from 'react-test-renderer';
import { mapStateToProps, Header } from 'plugins/log/components/Header';
import { themed } from 'utils/tests';

describe('plugins/log/components/Header', () => {
  describe('Header', () => {
    it('renders correctly', () => {
      const tree = renderer.create(
        themed(<Header
          start={jest.fn()}
          connected
          stop={jest.fn()}
          query=""
          lines="400"
          clearLogs={jest.fn()}
        />)
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('mapStateToProps', () => {
    it('should return the right stuff', () => {
      expect(mapStateToProps({ log: {
        connected: true,
        lines: 400,
        query: 'query',
      } })).toMatchSnapshot();
    });
  });
});
