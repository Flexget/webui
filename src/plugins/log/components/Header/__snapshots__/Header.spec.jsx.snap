// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plugins/log/components/Header Header renders correctly 1`] = `
.glamor-4 {
  display: block;
}

.glamor-0 {
  color: #757575;
}

.glamor-1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.glamor-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
}

.glamor-2 {
  margin-left: 1rem;
  margin-right: 1rem;
}

@media (min-width:600px) {
  .glamor-4 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: baseline;
    -webkit-box-align: baseline;
    -ms-flex-align: baseline;
    align-items: baseline;
  }
}

<div
  className="glamor-4"
>
  <div>
    <h2
      className="MuiTypography-root-1 MuiTypography-title-7"
    >
      Server Log
    </h2>
    <h3
      className="MuiTypography-root-1 MuiTypography-subheading-8 glamor-0"
    >
      Streaming
    </h3>
  </div>
  <div
    className="glamor-1"
  />
  <div
    className="glamor-3"
  >
    <div
      className="MuiFormControl-root-25 glamor-2"
      inputProps={
        Object {
          "onKeyPress": [Function],
        }
      }
      onBlur={[Function]}
      onFocus={[Function]}
      required={false}
    >
      <label
        className="MuiFormLabel-root-35 MuiInputLabel-root-29 MuiInputLabel-formControl-30 MuiInputLabel-animated-33"
        data-shrink={false}
        htmlFor="filter"
      >
        Filter
      </label>
      <div
        className="MuiInput-root-39 MuiInput-formControl-40 MuiInput-inkbar-41 MuiInput-underline-45"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <input
          autoComplete={undefined}
          autoFocus={undefined}
          className="MuiInput-input-48 MuiInput-inputSingleline-51"
          defaultValue={undefined}
          disabled={false}
          id="filter"
          name={undefined}
          onChange={[Function]}
          onKeyDown={undefined}
          onKeyUp={undefined}
          placeholder={undefined}
          readOnly={undefined}
          required={undefined}
          rows={undefined}
          type="text"
          value=""
        />
      </div>
      <p
        className="MuiFormHelperText-root-54"
      >
        Supports operators and, or, (), and "str"
      </p>
    </div>
  </div>
</div>
`;

exports[`plugins/log/components/Header mapStateToProps should return the right stuff 1`] = `
Object {
  "connected": true,
}
`;
